internal/server/rest.go

```diff
@@ -170,8 +170,50 @@ func createRestServer(ip string, port int, needToken bool) *http.Server {
 }
 
 type fileContent struct {
-	Name    string `json:"name"`
-	Content string `json:"content"`
+	Name     string `json:"name"`
+	Content  string `json:"content"`
+	FilePath string `json:"file"`
+}
+
+func (f *fileContent) Validate() error {
+	if f.Content == "" && f.FilePath == "" {
+		return fmt.Errorf("invalid body: content or FilePath is required")
+	}
+	if f.Name == "" {
+		return fmt.Errorf("invalid body: name is required")
+	}
+	return nil
+}
+
+func upload(file *fileContent) error {
+	err := getFile(file)
+	if err != nil {
+		return err
```

Comment:
Please try to cover these error condtions as well.

Reply:
ack